dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -0,0 +1,371 @@
+                                 "size or offset corresponding to the extension not found"));
+    DALI_ENFORCE(
+        component.offset % kBlockSize == 0,
+        IndexFileErrMsg(index_path, line, "tar offset is not a multiple of tar block size (",
+                        kBlockSize, "), perhaps the size value is exported before offset?"));
+    components_container.emplace_back(std::move(component));
+    samples_container.back().components.num++;
+  }
+
+  // Finishing up the SampleDesc
+  DALI_ENFORCE(samples_container.back().components.num,
+               IndexFileErrMsg(index_path, line, "no extensions provided for the sample"));
+}
+
+inline void ParseIndexFile(std::vector<SampleDesc>& samples_container,
+                           std::vector<ComponentDesc>& components_container,
+                           const std::string& index_path) {
+  std::ifstream index_file(index_path);
+
+  // Index Checking
+  std::string index_version;
+  DALI_ENFORCE(index_file >> index_version,
+               IndexFileErrMsg(index_path, 0, "no version signature found"));
+  DALI_ENFORCE(kCurrentIndexVersion == index_version,
+               IndexFileErrMsg(
+                   index_path, 0,
+                   "the version of the index file does not match the expected version (expected: ",
+                   kCurrentIndexVersion, " actual: ", index_version, ")"));
+
+  // Getting the number of samples in the index file
+  int64_t sample_desc_num_signed;
+  DALI_ENFORCE(index_file >> sample_desc_num_signed,
+               IndexFileErrMsg(index_path, 0, "no sample count found"));
+  DALI_ENFORCE(sample_desc_num_signed > 0,
+               IndexFileErrMsg(index_path, 0, "sample count must be positive"));
+  index_file.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+  const size_t sample_desc_num = sample_desc_num_signed;
+  samples_container.reserve(samples_container.size() + sample_desc_num);
+  for (size_t sample_index = 0; sample_index < sample_desc_num; sample_index++) {
+    ParseSampleDesc(samples_container, components_container, index_file, index_path,
+                    sample_index + 1);
+  }
+}
+
+}  // namespace wds
+}  // namespace detail
+
+inline std::string SupportedTypesListGen() {
+  std::stringstream out;
+  for (auto& dtype : detail::wds::kSupportedTypes) {
+    out << dtype << ',';
+  }
+  std::string out_str = out.str();
+  return out_str.substr(0, out_str.size() - (detail::wds::kSupportedTypes.size() > 0));
+}
+
+WebdatasetLoader::WebdatasetLoader(const OpSpec& spec)
+    : Loader(spec),
+      uris_(spec.GetRepeatedArgument<std::string>("uris")),
+      index_paths_(spec.GetRepeatedArgument<std::string>("index_paths")),
+      missing_component_behavior_(detail::wds::ParseMissingExtBehavior(
+          spec.GetArgument<std::string>("missing_component_behavior"))) {
+  DALI_ENFORCE(uris_.size() == index_paths_.size(),
+               "Number of uris does not match the number of index files");
```

Comment:
```suggestion
               "Number of data files does not match the number of index files");
```
I'm not sure if uri means much in this context.

Reply:
just the name of the variable, will change the error message